__tests__/rtt.test.js

```diff
@@ -0,0 +1,304 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY NORDIC SEMICONDUCTOR ASA "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY, NONINFRINGEMENT, AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL NORDIC SEMICONDUCTOR ASA OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
+ * TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+'use strict';
+
+const nRFjprog = require('../index.js');
+
+let device;
+let RTT = new nRFjprog.RTT();
+
+jasmine.DEFAULT_TIMEOUT_INTERVAL = 100000;
+
+describe('RTT', () => {
+    beforeAll(done => {
+        const programCallback = err => {
+            expect(err).toBeUndefined();
+
+            done();
+        };
+
+        const callback = (err, connectedDevices) => {
+            expect(err).toBeUndefined();
+            expect(connectedDevices.length).toBeGreaterThanOrEqual(1);
+            device = connectedDevices[0];
+
+            nRFjprog.program(device.serialNumber, "./__tests__/hex/rtt.hex", { }, programCallback);
+        };
+
+        nRFjprog.getConnectedDevices(callback);
+    });
+
+    describe('opens and close as expected', () => {
+        it('starts a rtt session', (done) => {
+            const stopCallback = (err) => {
+                expect(err).toBeUndefined();
+                done();
+            };
+
+            const startCallback = (err, down, up) => {
+                expect(err).toBeUndefined();
+                expect(down).toBeDefined();
+                expect(up).toBeDefined();
+
+                RTT.stop(stopCallback);
+            };
+
+            RTT = new nRFjprog.RTT();
+            RTT.start(device.serialNumber, {}, startCallback);
+        });
+
+        it('starts a rtt session when using correct control block location', (done) => {
+            const stopCallback = (err) => {
+                expect(err).toBeUndefined();
+                done();
+            };
+
+            const startCallback = (err, down, up) => {
+                expect(err).toBeUndefined();
+                expect(down).toBeDefined();
+                expect(up).toBeDefined();
+
+                RTT.stop(stopCallback);
+            };
+
+            RTT = new nRFjprog.RTT();
+            RTT.start(device.serialNumber, { controlBlockLocation: 0x200006E0 }, startCallback);
+        });
+
+        it('returns an error when wrong serialnumber', done => {
+            const startCallback = (err, down, up) => {
+                expect(err).toBeDefined();
+                expect(err).toMatchSnapshot();
```

Comment:
I'm not a fan of having the error definitions in a different file, but don't really care a lot.

Reply:
The snapshot feature is in my opinion a nice feature, though I see your point. Will not change it at this point in time.